### PR TITLE
[KIECLOUD-97] - Propagate OpenShiftStartupStrategy related changes to…

### PR DESCRIPTION
**Pull request by spolti**
… KIE APB

replace all occurrencies of dash by underscore

witht he current solution only the first dash is replace leading the hook
to fail where the WORKBANCH_SERVICE_NAME contains more than 1 dash in its
value.

Signed-off-by: Filippe Spolti <fspolti@redhat.com>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[KIECLOUD-XYZ] Subject`, `[RHDM-XYZ] Subject` or `[RHPAM-XYZ] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

### CODE REVIEW

File: jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh [hunk 1]
```diff
@@ -22,7 +22,7 @@ update_config_map() {
             -H 'Accept: application/json' \
             -H 'Content-Type: application/strategic-merge-patch+json' \
             -XPATCH \
-            https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/v1/namespaces/${namespace}/configmaps/myapp-kieserver -d ${payload} )
+            https://${KUBERNETES_SERVICE_HOST:-kubernetes.default.svc}:${KUBERNETES_SERVICE_PORT:-443}/api/v1/namespaces/${namespace}/configmaps/${KIE_SERVER_ID} -d ${payload} )
```
Comment (rhtevan) [inline jboss-kie-kieserver/added/launch/jboss-kie-kieserver-hooks.sh]: @spolti The KIE Server DC name is not necessarily to the same as KIE Server id due to one of the enhancements specified by this JIRA (https://issues.jboss.org/projects/JBPM/issues/JBPM-8150). However, we do mandate that all the OpenShift resources, such as DC, Service, Route and CM associated with one KIE Server uses exactly the same name, which could be a generated synthetic name like: 'myapp-kieserver-0'. Then, the actual KIE Server ID will be carried by a reserved label: 'services.server.kie.org/kie-server-id'. 
As a result, we may want to do something like this:
```
kieDCName=$(echo ${kieResponse:: -3} | python -c 'import json,sys;obj=json.load(sys.stdin);print (obj["items"][0]["metadata"]["name"])')
```
Then, pass that as extra argument to function update_config_map().

**Comment by sutaakar** [comment]
@spolti  Could you please add test coverage for this fix?

**Comment by spolti** [comment]
> @spolti Could you please add test coverage for this fix?

Not sure hwo to do that since we depend on k8s api, thoughts?

**Comment by rhtevan** [comment]
@sutaakar @spolti Since the script is dependent on a running k8s API server, it is not straightforward to construct a unit test case, like java code using jUnit, even with the help of BEHAVE. In addition, this is a short term fix. It will be replaced by a java component which will certainly include unit tests using f8's mock k8s server. For now, thorough review and manually test it out could be more achievable approach. Do you agree?

**Comment by ricardozanini** [comment]
On EAP Modules we created something like this: https://github.com/jboss-container-images/jboss-eap-modules/blob/master/os-eap-sso/tests/bats/hostname-discovery.bats

**Comment by sutaakar** [comment]
sounds ok to me

**Comment by spolti** [comment]
Let's not merge this yet, we can mock k8s requests as @ricardozanini suggested.

**Comment by spolti** [comment]
Unfortunately the suggestion made by Zanini will not work in this case since there is no way to break the jboss-kie-kieserver-hook script in smaller pieces to being able to perform the requests separately.
What we could to is write a little python or something which answers the desired response based on the uri.

**Pull request merged**